app/process/handler/consensus-message/consensus-message.go

```diff
@@ -1,3 +1,114 @@
 package consensusmessage
 
-// TODO: Consensus message event handler
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/golang/protobuf/proto"
+	"github.com/limechain/hedera-eth-bridge-validator/app/domain/repositories"
+	"github.com/limechain/hedera-eth-bridge-validator/app/persistence/message"
+	"github.com/limechain/hedera-eth-bridge-validator/config"
+	validatorproto "github.com/limechain/hedera-eth-bridge-validator/proto"
+	"github.com/limechain/hedera-watcher-sdk/queue"
+	log "github.com/sirupsen/logrus"
+	"strings"
+)
+
+type ConsensusMessageHandler struct {
+	repository     repositories.MessageRepository
+	validAddresses []string
+}
+
+func (cmh ConsensusMessageHandler) Recover(queue *queue.Queue) {
+	log.Println("Recovery method not implemented yet.")
+}
+
+func NewConsensusMessageHandler(r repositories.MessageRepository) *ConsensusMessageHandler {
+	return &ConsensusMessageHandler{
+		repository:     r,
+		validAddresses: config.LoadConfig().Hedera.Handler.ConsensusMessage.Addresses,
```

Comment:
It would be good to validate configuration addresses (regex checks).

Reply:
Can we do that in the next PR?